### PR TITLE
magento/magento2#10353 : Cant put id or data attribute on <img> tags …

### PR DESCRIPTION
**Pull request by benathomeofchampions**
…in the M2 wysiwyg.

Fixes issue whereby an ID attribute is added to an  via the WYSIWYG, and a proceeded base64 decode of the ID causes an exception.

This introduces a new method Base64.isBase64() and performs a check before trying to get an widget code from the ID.

### Description (*)
Manually adding an ID attribute to an  tag via the WYSIWYG / editor results in the ID being parsed for widget code using the window.atob, which throws an exception if the string is not valid base64.

### Fixed Issues (if relevant)

1. magento/magento2#10353: Cant put id or data attribute on  tags in the M2 wysiwyg.

### Manual testing scenarios (*)

1. Edit a page and add an image
2. Save the page
3. Edit the markup and add an ID attribute to the 
4. Save the page, without seeing an exception.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=23302) <br/>All committers have signed the CLA.

**Comment by torhoehn** [comment]
@benathomeofchampions Please fix failing Functional and Static Tests.

**Comment by sidolov** [comment]
@magento run all tests

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-6008](https://jira.corp.magento.com/browse/ENGCOM-6008) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by engcom-Golf** [comment]
@magento run ALL tests

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
Hi @benathomeofchampions could you please sign in new cla ?

File: lib/web/mage/adminhtml/wysiwyg/tiny_mce/plugins/magentowidget/editor_plugin.js [hunk 1]
```diff
@@ -170,7 +171,7 @@ define([
                         var attributes = wysiwyg.parseAttributesString(match[2]),
                             widgetCode;
 
-                        if (attributes.id) {
+                        if (attributes.id && Base64.isBase64(attributes.id)) {
                             widgetCode = Base64.idDecode(attributes.id);
```
Comment (engcom-Golf) [inline lib/web/mage/adminhtml/wysiwyg/tiny_mce/plugins/magentowidget/editor_plugin.js:175]: Hi @benathomeofchampions thank you for your contribution, after small fixes where you forgot to add your method in min.js file, we found that tests fail with insert widget in WYSIWYG edit could you please see failed test and fix them? the problem is that when you want to add 2 widgets with catalog product list the second widget is not decoding.
DeepinScreenshot_select-area_20191009110735
 link to the test result https://testing-service.magento-community.engineering/reports/magento/magento2/pull/23302/444d44345ef23c8782e9694b43b9bd84/Functional/allure-report-ce/index.html#suites/8187718e2fd144987673ef34ee62b41b/47d4a5f00507d5d8/

**Review by engcom-Golf — CHANGES_REQUESTED** [review]
Hi, @benathomeofchampions thank you for pull request, please see my review comments.

**Comment by krzksz** [comment]
Closing this merge request due to inactivity. Feel free to reopen if you wish to continue working on this issue.

**Pull request closed without merging**